### ISSUE TITLE
Update version of feature when plugin evolves

### ISSUE THREAD

**Issue by vladdu** [inventage/version-tiger#36]
I think that when the projects are Eclipse plugins and features, it would be useful to automatically update a feature's version when one of its plugins get updated.

This may be a bit tricky because if one updates versions more often than before a release, the feature might end with a version several steps higher than strictly needed. To do it right, one would need to use the versions for the latest release as baseline.

**Comment by b8** [comment]
Yes, this would be a great enhancement. I had the idea to add rules to a pom to specify which projects should also be updated. I think this feature is one of the next things to do.

**Comment by vladdu** [comment]
I will try to submit a PR for this. 

I suppose on could navigate the dependencies in the poms and select even upstream modules. 

For Eclipse plugins, the dependencies feature->plugins and feature->feature are not usually explicit in the pom, so something must be done automagically. The better solution is the one you suggested, having rules to specify dependencies that aren't defined in the pom. I'm not sure what a good implementation would be (the rules might need associated code, for example in a workspace with multiple feature projects, detecting the right one that contains a plugin requires scanning feature.xml), so I think that to start with  it might be good enough with a specialized solution.
